pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -1800,13 +1800,12 @@ impl ResidentLayer {
         use LayerKind::*;
 
         let owner = &self.owner.0;
+        owner
+            .access_stats
+            .record_access(LayerAccessKind::KeyIter, ctx);
```

Comment:
Why was this moved? It's not a big issue but it was more correct to do it after the kind check.

Reply:
Moved it back in 010b18aeb -- an earlier version of this change included making load_keys work for image layers..